### ISSUE TITLE
Fedora 25 not booting

### ISSUE THREAD

**Issue by ukasuviok** [reynhout/chrx#36]
The install for Fedora 25 works without a problem, but the boot process times out while trying to look for /dev/mmcblk0p7 and then drops to a rescue shell.  I've tried to install it multiple times and actually reset the Chromebook to have the script recreate the partitions.  Still no luck though.  I haven't had any issues installing Ubuntu or Gallium.  Hardware is an Asus 302 Flip.

Warning:  Could not boot.
Warning:  /dev/mmcblk0p7 does not exist

**Comment by reynhout** [comment]
@ukasuviok 

Hi, would you be able to post the logfile from the install?

At the end of the installation, instead of hitting return to reboot, press Ctrl+C and then run:

```
cat /var/tmp/chrx/chrx-install.log | curl -sF c=@- https://ptpb.pw | grep url:
```

**Comment by ukasuviok** [comment]
url: https://ptpb.pw/juaV

**Comment by reynhout** [comment]
I don't see anything terribly wrong there.

When you boot to a rescue image, does `/dev/mmcblk0p7` show up in the output of `lsblk`?

**Comment by ukasuviok** [comment]
Now it just hangs during boot and won't drop me into the rescue shell.  I did notice that when I boot into a Fedora live image and try to manually partition the drive I can see everything except mmcblk0p7.

**Comment by reynhout** [comment]
Hm, I have no good leads then. @jedigo any ideas?

@ukasuviok Would it work for you to install F24 and then update to F25?

**Comment by ukasuviok** [comment]
Is the script supposed to download the 25 release docker base when specifying -r 24?  That seems to be what it's doing, but I'll keep the install going.

**Comment by reynhout** [comment]
@ukasuviok I think I remember noticing that the pre-install image will be the "latest", no matter which version will ultimately be installed.

**Comment by ukasuviok** [comment]
The same thing happens on 24 as well.

url: https://ptpb.pw/PQMX

**Comment by reynhout** [comment]
@ukasuviok OK, thanks for the report and the update.

I don't have any Skylake machines to test Fedora on, so I don't know whether I'll be able to repro the problem, but I'll try on another device with eMMC storage at least.

**Comment by ukasuviok** [comment]
Thanks!

**Comment by hmrubin** [comment]
I can confirm that this bug occurs when installing fedora on a broadwell chromebook. Acer c910-c37p yuna

**Comment by xKim** [comment]
I have the same problem. Though I am trying to install fedora on a acer c735. It worked on the day that was available on the first day. I have a intel bay trail cpu. I switched to the terminal mode while booting and it was like: time/infinite. I have the same problem but im on a different hardware. I think it is an update issue or something.

**Comment by reynhout** [comment]
It looks pretty certain that there has been a recent change in the Fedora images that chrx downloads. I suspect the change occurred within the last few days, because these reports are all arriving at the same time, and there have been no substantive changes to chrx in the same time period.

@xKim ChrUbuntu stopped working when Ubuntu switched to systemd. I don't follow Fedora, so I don't know if there are any obvious architectural candidates here.

@jedigo wrote the Fedora code for chrx, and aside from testing back when it first landed, I haven't had occasion to work on it at all. I'm hoping he'll have some time to revisit it.

For now I'd recommend installing another distro. GalliumOS is the best-tested.

**Comment by jedigo** [comment]
Hi, i will investigate this ASAP, I know there is an issue booting with kernel 4.9 on baytrail - I need to open a bug upstream, but in the meantime i can set the install to ignore kernel 4.9 when it installs / updates. I will make a pull request for this later today or tomorrow.

I will try and investigate the installation issues to see if there is any other problems, but it will probably be a full week at the earliest before I can spend enough time to troubleshoot.

**Comment by reynhout** [comment]
@jedigo Great, thank you. If you think of it, please link the bug report for the 4.9 kernel on baytrail, it might be helpful for others too.

**Comment by jedigo** [comment]
Hi all, the issue is actually the latest update kernel meta package not installing kernel modules.

I have fixed this in the pull request, and also caused the original kernel for the OS release, in case anyone else has issues with later kernels, like my issues with 4.9/4.10 breaking keyboard support - I am not sure if this is just for my laptop, or for more Braswell Chromebooks.

**Comment by hmrubin** [comment]
@jedigo so we are good to install it? 
Thank you for all of yout had work. I always love experimenting with other operating systems

**Comment by reynhout** [comment]
@jedigo Thank you! One question on the PR before merging (in a comment there).

The keyboard issue on Braswell models with kernel 4.9 is **fixed** in GalliumOS, but not in other distros.

I don't know whether a fix will percolate into future 4.9 (or 4.10, etc) mainline kernels for the other distros to pick up.

**Comment by reynhout** [comment]
This should be fixed, thanks to @jedigo. Updated code has been pushed to chrx.org.
